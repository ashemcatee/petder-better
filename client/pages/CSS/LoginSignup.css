body {
    background: url("../../public/dog.jpg");
    background-size: cover;
    background-repeat: none;
    background-attachment: fixed;
}

.wrapper {
    display: flex;
    justify-content: end;
    margin-right: 50px;
}

.form-box{
    position:relative;
    margin-left: 600px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 600px;
    height: 65vh;
    overflow: hidden;
    padding-left: 10px;
    padding-right: 10px;
    margin-top: 200px;
    background-color: rgba(255, 255, 255, 0.25);
}

.register-container {
    width: 500px;
    display: flex;
    flex-direction: column;
    margin: 0;
    padding: 20px;
}

.oneLine {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 20px;
    margin-bottom: 20px;
    margin: 0;
}

.oneLine span {
    color: #45b3e7;
    font-size: small;
    padding: 10px 0;
    display: flex;
    justify-content: center;
    cursor: pointer;
    margin: 0;
}

header {
    color: black;
    font-size: 30px;
    text-align: center;
    padding: 0;
}
.input-box input{
    margin-top: 0px;
}
.input-field {
    font-size: 15px;
    background:  #e6f1f6;
    color: black;
    height: 50px;
    width: 100%;
    padding: 0 10px 0 45px;
    margin-bottom: 20px;
    border: none;
    border-radius: 30px;
    transition: .2s ease;

}
.input-field:hover, .input-field:focus{
    background:rgba(255, 255, 255, 0.25);
}
::-webkit-input-placeholder {
    color: black;
}

#submit {
    font-size: 15px;
    font-weight: 500;
    color: black;
    height: 45px;
    width: 100%;
    border: none;
    border-radius: 30px;
    background: #45b3e7;
    cursor: pointer;
}

#submit:hover{
    background:#89cff0;
    transition: .2s ease;
}

.error-message {
    padding-bottom: 10px;
}

.loginsignup-agree {
    display: flex;
    flex-direction: row;
    font-size: 10px;
    margin-top: 20px;
    font-weight: lighter;
    margin-left: 10px;
    gap: 10px;
}

.oneLine {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 10px;
}

.oneLine p {
    margin: 0;
    font-size: 14px;
}

.oneLine span {
    color: #007BFF;
    cursor: pointer;
    text-decoration: underline;
}

.input-box {
    margin-bottom: 20px;
}

.error-message {
    color: red;
    margin-top: 10px;
}

.loginsignup-agree {
    display: flex;
    align-items: center;
    margin-top: 20px;
}

.loginsignup-agree input[type="checkbox"] {
    margin-right: 10px;
}

.loginsignup-agree p {
    margin: 0;
    font-size: 12px;
}

input::placeholder {
    color: rgba(128, 128, 128, 0.741);
}
  
@media (max-width: 1024px) {
    .wrapper {
        justify-content: center; 
        
    }
    .form-box {
        margin: 0 auto; 
        margin-top: 200px;
    }
}

@media (min-width: 1025px) {
    .wrapper {
        justify-content: flex-end; 
    }
    .form-box {
        margin-right: 50px; 
    }
}

