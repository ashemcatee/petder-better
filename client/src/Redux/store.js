import authReducer from "./AuthReducer";
import petReducer from "./PetReducer";
import { configureStore } from '@reduxjs/toolkit';

const store = configureStore({
    reducer: {
        auth: authReducer,
        pets: petReducer,
    }
});

export default store;