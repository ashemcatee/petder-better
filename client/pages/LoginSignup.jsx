import React, {useState, useEffect} from 'react';
import './CSS/LoginSignup.css';
import { useSelector, useDispatch } from 'react-redux'; 
import { setAuthView, loginSuccess, signupSuccess } from '../src/Redux/action';


const LoginSignup = () => {
    const dispatch = useDispatch();

    //get current auth view: "Login or SignUp"
    const authView = useSelector(state => state.auth.authView);

    // recording state of error
    const [error, setError] = useState('');
    // recording formData that will send to server
    const [formData, setFormData] = useState({
        "username": "",
        "password": "",
        "email": "",
    });

    const changeHandler = (e) => {
        setFormData({...formData, [e.target.name]: e.target.value})
    };

    // login
    const login = async() => {
        let responseData;
        console.log('formData', formData);
        await fetch('http://localhost:3000/api/login', {
            method: "POST",
            headers:{
                Accept: "application/form-data",
                "Content-Type": "application/json",
            },
            body: JSON.stringify(formData),
        })
        .then((response) => response.json())
        .then((data) => {responseData = data})
        .catch(err => {
            setError('Login failed. Please try again.');  // Handle fetch error
        });

        if (responseData.success) {
            window.location.replace("/profile");
            dispatch(setAuthView(loginSuccess));
        } else {
            setError('Invalid username or password.');  // Handle response error
        }
    } 

    // signup
    const Signup = async () => {
        try {
            const response = await fetch('http://localhost:3000/api/signup', {
                method: "POST",
                headers: {
                    Accept: "application/json",
                    "Content-Type": "application/json",
                },
                body: JSON.stringify(formData),
            });
    
            if (!response.ok) {
                throw new Error('Network response was not ok'); // Throws an error if response is not OK
            }
    
            const responseData = await response.json();
            if (responseData.success) {
                window.location.replace("/profile");
                dispatch(setAuthView(signupSuccess));
            } else {
                setError('Failed to create an account. Please check your details and try again.');
            }
        } catch (error) {
            console.error('Signup Error:', error);
            setError('Signup failed. Please try again.');
        }
    }

    return (
        <div className="wrapper">
             <div className="form-box">
                <div className="register-container" id="register">
                        { authView == "SignUp"? (
                        <div className="top">
                            <div className="oneLine">
                                <p>Already Have an account</p>
                                <span onClick={()=>{dispatch(setAuthView("Login"))}}>Login</span>
                            </div>
                            <header>Sign Up</header>
                        </div>
                        ):(<div className="top">
                        <div className="oneLine">
                            <p>Create a new Account</p>
                            <span onClick={()=>{dispatch(setAuthView("SignUp"))}}>Sign Up</span>
                        </div>
                        <header>Login</header>
                        </div>)}
                        
                    <div className="input-box">
                        <input 
                        type="text" 
                        className="input-field" 
                        placeholder="Your Name" 
                        name="username"
                        value={formData.username} 
                        onChange={changeHandler}/>
                    </div>
                    {authView == "SignUp"?(
                        <div className="input-box">
                        <input 
                        type="text" 
                        className="input-field" 
                        placeholder="Email"
                        name="email"
                        value={formData.email} 
                        onChange={changeHandler}/>
                    </div>
                    ):(<></>)}
                    <div className="input-box">
                        <input type="text" 
                        className="input-field" 
                        placeholder="Password"
                        name="password"
                        value={formData.password} 
                        onChange={changeHandler}/>
                    </div>
                    {error && (
                    <div className="error-message" style={{ color: 'red' }}>
                        {error}
                    </div>
                    )}

                    <button className="input-box" id="submit" onClick={() => {
                        authView == "SignUp" ? Signup() : login();
                    }}>Continue</button>

                    <div className="loginsignup-agree">
                        <input type="checkbox" name="" id="" />
                        <p>By Continuing, I agree to the terms of the use & privacy policy</p>
                    </div>
                </div>
            </div>
        </div> 
    )
}

export default LoginSignup;