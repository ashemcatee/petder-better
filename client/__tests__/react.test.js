import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LoginSignup from '../pages/LoginSignup.jsx';
import { Provider } from 'react-redux';
import store  from '../src/Redux/store.js';
import { configureStore } from '@reduxjs/toolkit';

describe('Unit Testing react components', () => {

describe('signup button', () => {
    test('renders signup component when signup button is clicked', () => {
  const { getByText, queryByText } = render(
  <Provider store={store}>
    <LoginSignup />
  </Provider>);

  // simulate click event
  const signupButton = getByText('SignUp');
  fireEvent.click(signupButton);

  expect(queryByText('Sign Up')).toBeInTheDocument();
});
})})


// import React from 'react';
// import { render, fireEvent } from '@testing-library/react';
// import { Provider } from 'react-redux';
// import { createStore } from 'redux';
// import LoginSignup from './LoginSignup.jsx';
// import rootReducer from '../src/Redux/reducer'; // import your root reducer

// test('renders signup component when signup button is clicked', () => {
//   // Create a mock store
//   const store = createStore(rootReducer);

//   // Render the component wrapped in a Provider with the mock store
//   const { getByText, queryByText } = render(
//     <Provider store={store}>
//       <LoginSignup />
//     </Provider>
//   );

//   // Rest of your test code...
// });