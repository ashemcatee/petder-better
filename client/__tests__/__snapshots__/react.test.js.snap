// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="form-box"
    >
      <div
        class="register-container"
        id="register"
      >
        <div
          class="top"
        >
          <div
            class="oneLine"
          >
            <p>
              Create a new Account
            </p>
            <span>
              Sign Up
            </span>
          </div>
          <header>
            Login
          </header>
        </div>
        <input
          class="input-field"
          name="username"
          placeholder="Your Name"
          type="text"
          value=""
        />
        <input
          class="input-field"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          class="input-box"
          id="submit"
        >
          Continue
        </button>
        <div
          class="loginsignup-agree"
        >
          <input
            id="termsCheckbox"
            type="checkbox"
          />
          <label
            for="termsCheckbox"
          >
            By Continuing, I agree to the terms of the use & privacy policy
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
